### ISSUE TITLE
Avoid Full Backup on Startup

### ISSUE THREAD

**Issue by vlerenc** [gardener/etcd-backup-restore#150]
**Feature (What you would like to be added):**
It would be great to avoid a full backup on startup when the cluster is anyways busy to get going. In this critical and time sensitive phase, a full backup puts ETCD under additional pressure and consumes precious network bandwidth.

**Motivation (Why is this needed?):**
We may hinder ourselves in getting the cluster operational as fast as possible, which is especially important while we run single-pod ETCDs. If ETCD get rescheduled due to an update, we depend on it starting up as fast as possible.

**Approach/Hint to the implement solution (optional):**
From an offline discussion, we did that (full backup on startup), because we wanted to be sure to have a guaranteed full backup even if ETCD fails again and again. However, not only may this cause a failure itself (too much CPU, memory and network pressure), it also slows everything down and may result in repeated full backups. We should rather wait for a couple of minutes and execute it only if it didn't happen in the past 24h (or execute it right away if it didn't happen for 48h). A marker file on disk or checking the backup bucket could provide the necessary insight when the last full backup was taken. This way, we would avoid full backups on startup, don't add additional stress in this phase, but still make sure, with the given time windows, to have full backups.

**Comment by shreyas-s-rao** [comment]
@vlerenc
> We should rather wait for a couple of minutes and execute it only if it didn't happen in the past 24h (or execute it right away if it didn't happen for 48h)
Instead of deferring full snapshot to be done after a couple of minutes, we can probably take a delta snapshot on startup (for all the events from the previous delta snapshot on the blob store), or take a full snapshot if there wasn't one in the last 24 hours, as you suggested. The delta snapshot won't stress etcd and network as much as a full snapshot. Also, this helps us handle the scenario where the previous instance of etcd had gracefully shut down just before taking a delta snapshot (so there's almost 5 minutes of events accumulated and not backed up to the blob store), and after etcd gets rescheduled, within those couple of minutes before we take a full snapshot there might be a possible data corruption, which could lead to a loss of 7 minutes of events, which might affect our SLAs. 

Basically, initial delta snapshot is more resource-efficient than initial full snapshot, and also safer (in terms of data and SLA) than directly taking full snapshot after a few minutes. WDYT?

**Comment by vlerenc** [comment]
Sounds like a very good proposal. Yes, thank you!
